ExampleObjc/ViewController.m

```diff
@@ -0,0 +1,80 @@
+//
+//  ViewController.m
+//  ExampleObjc
+//
+//  Created by Dave Hardiman on 07/03/2016.
+//  Copyright © 2016 Ordnance Survey. All rights reserved.
+//
+
+#import "ViewController.h"
+@import Routing;
+@import MapKit;
+
+@interface ViewController ()<MKMapViewDelegate>
+@property (weak, nonatomic) IBOutlet MKMapView *mapView;
+@property (strong, nonatomic) OSRoutingService *routingService;
+@property (strong, nonatomic) NSMutableArray *tappedPoints;
+@end
+
+@implementation ViewController
+
+- (NSString *)apiKey {
+    return [NSString stringWithContentsOfURL:[NSBundle.mainBundle URLForResource:@"APIKEY" withExtension:nil]
+                                    encoding:NSUTF8StringEncoding
+                                       error:nil];
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.routingService = [[OSRoutingService alloc] initWithApiKey:self.apiKey vehicleType:[OSRoutingService carVehicleType] crs:nil];
+    self.mapView.delegate = self;
+    self.tappedPoints = NSMutableArray.array;
+
+    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(mapTapped:)];
+    [self.mapView addGestureRecognizer:tap];
+}
+
+#pragma mark - Map Tapped
+- (void)mapTapped:(UITapGestureRecognizer *)tap {
+    if (tap.state != UIGestureRecognizerStateEnded) {
+        return;
+    }
+    CGPoint touchPoint = [tap locationInView:self.mapView];
+    CLLocationCoordinate2D touchCoordinate = [self.mapView convertPoint:touchPoint toCoordinateFromView:self.mapView];
+    MKPointAnnotation *pointAnnotation = [[MKPointAnnotation alloc] init];
+    pointAnnotation.coordinate = touchCoordinate;
+    [self.mapView addAnnotation:pointAnnotation];
+    [self.tappedPoints addObject:[NSValue os_valueWithCoordinate:touchCoordinate]];
+    [self updateRoute];
+}
+
+- (void)updateRoute {
+    [self.routingService routeBetweenCoordinates:self.tappedPoints completion:^(OSRoute *_Nullable route, NSError *_Nullable error) {
+        if (error) {
+            NSLog(@"%@", error);
+            return;
+        }
+        [self displayRoute:route];
+    }];
+}
+
+- (void)displayRoute:(OSRoute *)route {
+    NSArray *coordinateValues = route.coordinateValues;
+    NSInteger numberOfPoints = coordinateValues.count;
+    CLLocationCoordinate2D *coordinates = malloc(sizeof(CLLocationCoordinate2D) * numberOfPoints);
+    for (NSInteger idx = 0; idx < numberOfPoints; idx++) {
+        coordinates[idx] = [coordinateValues[idx] os_coordinateValue];
+    }
+    MKPolyline *line = [MKPolyline polylineWithCoordinates:coordinates count:numberOfPoints];
+    [self.mapView addOverlay:line];
+    free(coordinates);
```

Comment:
I'd rather not go down that route as it then ties the framework to MapKit, which I'm trying to avoid. Only reason for using `malloc`/`free` is to avoid this warning:
![screen shot 2016-03-07 at 09 56 58](https://cloud.githubusercontent.com/assets/272213/13573062/a042c8ae-e475-11e5-9264-35cf45bebdfa.png)
which manifested itself when the code looked like
![screen shot 2016-03-07 at 09 58 20](https://cloud.githubusercontent.com/assets/272213/13573072/a95b2a9e-e475-11e5-9f5e-fd455bc13eb0.png)
and I had no idea how to squash the warning.